features/userlist/api/src/main/kotlin/io/element/android/features/userlist/api/UserListDataStore.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.userlist.api
+
+import io.element.android.libraries.matrix.ui.model.MatrixUser
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import javax.inject.Inject
+
+class UserListDataStore @Inject constructor() {
+
+    private val selectedUsers: MutableStateFlow<List<MatrixUser>> = MutableStateFlow(emptyList())
+
+    fun selectUser(user: MatrixUser) {
+        selectedUsers.tryEmit(selectedUsers.value.plus(user))
```

Comment:
Do we want to check that a user is not added twice here?

Reply:
The check was already done in the presenter, but it probably makes sense to move it there: https://github.com/vector-im/element-x-android/pull/320/commits/2748726d3bf6888cf7cc4e1ffba1c778f072f7e0